Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -104,6 +104,39 @@ static void assertedElementSetState(GstElement* element, GstState desiredState)
     }
 }
 
+void AppendPipeline::configureOptionalDemuxer()
+{
+    String elementClass = unsafeSpan(gst_element_get_metadata(m_demux.get(), GST_ELEMENT_METADATA_KLASS));
+    // We try to detect special cases of demuxers that have a single static src pad, such as id3demux.
+    GRefPtr<GstPad> demuxerSrcPad = adoptGRef(gst_element_get_static_pad(m_demux.get(), "src"));
+    auto classifiers = elementClass.split('/');
+    if (classifiers.contains("Demuxer"_s) && !demuxerSrcPad) {
```

Comment:
From what I see if `demuxerSrcPad` is not `nullptr` there is no need to create a string and split it. This could probably be improved.

Reply:
Yes, but when it's not, you are creating a string and splitting it for no use. I'd refactor this code.

Reply:
Explaining aloud to myself my understanding of the comments, so I don't forget when I come back at this in one week:
- Evaluate `!demuxerSrcPad as first element in short-circuit (or alone) in the if
- Move elementClass and classifiers initialization inside the branch, or even inside the condition
- Make sure that the `else` code runs in the `!!demuxerSrcPad` case and also in case the element is not a "Demuxer".

Checking only `if (demuxerSrcPad)` and swapping the then and else branches could do the trick. If the element doesn't have a "src" pad (new else branch) and is not a demuxer, then we can just assert or something, I guess, because that would be an unsupported case (for the limited set of formats that AppendPipeline can support: webm, mp4, raw mpeg or aac stream).